### PR TITLE
tests/socket_zep: fix automatic test

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The automatic test for socket ZEP was broken because it did not run on Murdock. This PR attempts to fix this:

 - use the API as expected by the Radio HAL
 - supply a EUI-64 so the `native` instance always uses the same address
 - switch to IPv4 as Murdock does not support IPv6. This allows to run the test in CI

### Testing procedure

    make -C tests/socket_zep test

should now work again.

### Issues/PRs references

fixes https://github.com/RIOT-OS/RIOT/pull/16946#issuecomment-995010366

### CODE REVIEW

File: tests/socket_zep/main.c [hunk 1]
```diff
@@ -116,7 +104,6 @@ int main(void)
     _main_pid = thread_getpid();
 
     test_init();
-    test_send__iolist_NULL();
```
Comment (kfessel) [inline tests/socket_zep/main.c:119]: Why did you remove this part of the test ?

**Review by kfessel — COMMENTED** [review]
Changes seem sound,
Test in CI was run.
https://ci.riot-os.org/RIOT-OS/RIOT/17432/79891e93181dca6fc2f2086dce0e49fec2566499/output/compile/tests/socket_zep/native:gnu.txt

Just a small question

File: tests/socket_zep/main.c (hunk 1 above)
Comment (benpicco) [reply tests/socket_zep/main.c:119]: With the radio HAL sending a `NULL` buffer will no longer result in any data being transmitted. Previously this would send a header-only packet (and this test would verify that).

Comment (kfessel) [reply tests/socket_zep/main.c:119]: maybe:

    Elements of this list may have iolist_t::iol_size == 0 and (in this case only) iolist_t::iol_data == 0

should be tested instead

Comment (benpicco) [reply tests/socket_zep/main.c:119]: I think this makes no sense. We are on layer 2 here, so for a real device this would mean no data on the bus / air.
Only with the (old) ZEP implementation this would write the ZEP header (that wraps the IEEE 802.15.4 frame).

Mind you that we are using the sub-mac now which does make assumptions about the contents being transmitted (namely them being a IEEE 802.15.4 frame) - we'd now need to add checks there, e.g in `sys/net/link_layer/ieee802154/submac.c:414` where it checks if the ACK_REQ bit is set - for a case that will never happen in reality.

**Review by kfessel — APPROVED** [review]
changes are sound, test ran by murdock,  removal of unneeded test explained -> question resolved
